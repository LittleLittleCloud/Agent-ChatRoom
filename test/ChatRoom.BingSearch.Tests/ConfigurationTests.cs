using System.Text.Json;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using FluentAssertions;
using Json.Schema;
using Json.Schema.Generation;
using Xunit;

namespace ChatRoom.WebSearch.Tests
{
    public class ConfigurationTests
    {
        [Fact]
        [UseReporter(typeof(DiffReporter))]
        [UseApprovalSubdirectory("ApprovalTests")]
        public void VerifyConfigurationSchema()
        {
            var schema = new JsonSchemaBuilder()
                .FromType<WebSearchConfiguration>()
                .Build();

            var schemaFileName = "chatroom_web_search_configuration_schema.json";
            var schemaFilePath = Path.Join("template", "chatroom.websearch", schemaFileName);
            var schemaFile = File.ReadAllText(schemaFilePath);

            var json = JsonSerializer.Serialize(schema, new JsonSerializerOptions { WriteIndented = true });

            Approvals.Verify(json);
            schemaFile.Should().BeEquivalentTo(json);
        }
    }
}
