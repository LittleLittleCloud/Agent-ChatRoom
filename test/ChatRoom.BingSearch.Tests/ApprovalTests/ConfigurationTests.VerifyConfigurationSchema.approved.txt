{
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Name of the bing search agent, default is \u0027bing-search\u0027"
    },
    "system_message": {
      "type": "string",
      "description": "System message, default is \u0027You are a Bing search agent. You can search the web using Bing search engine.\u0027"
    },
    "description": {
      "type": "string",
      "description": "Agent description, default is \u0027I am a Bing search agent. I can search the web using Bing search engine.\u0027"
    },
    "bing_api_key": {
      "type": "string",
      "description": "Bing API key, will use $env:BING_API_KEY if not provided"
    },
    "openai_config": {
      "type": "object",
      "properties": {
        "llm_type": {
          "$ref": "#/$defs/lLMType"
        },
        "openai_api_key": {
          "$ref": "#/$defs/string"
        },
        "openai_model_id": {
          "$ref": "#/$defs/string1"
        },
        "azure_openai_endpoint": {
          "$ref": "#/$defs/string2"
        },
        "azure_openai_key": {
          "$ref": "#/$defs/string3"
        },
        "azure_openai_deploy_name": {
          "$ref": "#/$defs/string4"
        },
        "third_party_llm_endpoint": {
          "$ref": "#/$defs/string5"
        },
        "third_party_llm_key": {
          "$ref": "#/$defs/string6"
        },
        "third_party_llm_model_id": {
          "$ref": "#/$defs/string7"
        }
      },
      "description": "OpenAI configuration"
    }
  },
  "$defs": {
    "lLMType": {
      "enum": [
        "AOAI",
        "OpenAI",
        "ThirdParty"
      ],
      "description": "LLM provider, default is Azure OpenAI"
    },
    "string": {
      "type": "string",
      "description": "OpenAI API key, default is $env:OPENAI_API_KEY"
    },
    "string1": {
      "type": "string",
      "description": "OpenAI model ID, default is gpt-3.5-turbo"
    },
    "string2": {
      "type": "string",
      "description": "Azure OpenAI endpoint, default is $env:AZURE_OPENAI_ENDPOINT"
    },
    "string3": {
      "type": "string",
      "description": "Azure OpenAI key, default is $env:AZURE_OPENAI_API_KEY"
    },
    "string4": {
      "type": "string",
      "description": "Azure OpenAI deploy name, default is $env:AZURE_OPENAI_DEPLOY_NAME"
    },
    "string5": {
      "type": "string",
      "description": "Third-party LLM provider endpoint, default is $env:THIRD_PARTY_LLM_ENDPOINT"
    },
    "string6": {
      "type": "string",
      "description": "Third-party LLM provider key, default is $env:THIRD_PARTY_LLM_KEY"
    },
    "string7": {
      "type": "string",
      "description": "Third-party LLM provider model ID, default is $env:THIRD_PARTY_LLM_MODEL_ID"
    }
  }
}